pkg/ipc/ipc.go

```diff
@@ -645,7 +645,14 @@ func makeCommand(pid int, bin []string, config *Config, inFile, outFile *os.File
 	if err := cmd.Start(); err != nil {
 		return nil, fmt.Errorf("failed to start executor binary: %v", err)
 	}
+	c.exited = make(chan error, 1)
 	c.cmd = cmd
+	go func(c *command) {
+		err := c.cmd.Wait()
+		c.exited <- err
+		close(c.exited)
+		rp.SetDeadline(time.Now())
```

Comment:
Perhaps `time.Now().Add(n*time.Second)`? This is not expected to be needed frequently, but we can fail to read out the actual error message if we abort reads immediately.
Also a comment here would be good.